diff --git a/tools/create_entry_points.py b/tools/create_entry_points.py
index ced34f23b..91926614b 100755
--- a/tools/create_entry_points.py
+++ b/tools/create_entry_points.py
@@ -41,10 +41,9 @@ def main():
   sh_file = os.path.join(tools_dir, 'run_python.sh')
   bat_file = os.path.join(tools_dir, 'run_python.bat')
   for entry_point in entry_points:
-    if os.path.exists(os.path.join(root_dir, entry_point)):
-        os.remove(os.path.join(root_dir, entry_point))
     shutil.copy2(sh_file, os.path.join(root_dir, entry_point))
-    shutil.copy2(bat_file, os.path.join(root_dir, entry_point) + '.bat')
+    if not entry_point.startswith('tools'):
+      shutil.copy2(sh_file, os.path.join(os.environ['ED'], 'usr/bin', entry_point))
 
 
 if __name__ == '__main__':
diff --git a/tools/run_python.sh b/tools/run_python.sh
index b9fcf9532..de6e1b60b 100755
--- a/tools/run_python.sh
+++ b/tools/run_python.sh
@@ -9,21 +9,4 @@
 # To modify this file, edit `tools/run_python.sh` and then run
 # `tools/create_entry_points.py`
 
-if [ -z "$PYTHON" ]; then
-  PYTHON=$EMSDK_PYTHON
-fi
-
-if [ -z "$PYTHON" ]; then
-  PYTHON=$(which python3 2> /dev/null)
-fi
-
-if [ -z "$PYTHON" ]; then
-  PYTHON=$(which python 2> /dev/null)
-fi
-
-if [ -z "$PYTHON" ]; then
-  echo 'unable to find python in $PATH'
-  exit 1
-fi
-
-exec "$PYTHON" "$0.py" "$@"
+GENTOO_PYTHON "GENTOO_PREFIX/usr/GENTOO_LIB/emscripten/$(basename $0).py" "$@"
